app/src/main/java/org/simple/clinic/widgets/CheckBoxWithSuppressibleListener.kt

```diff
@@ -0,0 +1,22 @@
+package org.simple.clinic.widgets
+
+import android.content.Context
+import android.support.v7.widget.AppCompatCheckBox
+import android.util.AttributeSet
+
+class CheckBoxWithSuppressibleListener(context: Context, attrs: AttributeSet) : AppCompatCheckBox(context, attrs) {
+
+  private var listener: OnCheckedChangeListener? = null
+
+  override fun setOnCheckedChangeListener(listener: OnCheckedChangeListener?) {
+    super.setOnCheckedChangeListener(listener)
+    this.listener = listener
+  }
+
+  fun runWithoutListener(runner: () -> Unit) {
```

Comment:
What do you think about calling this `block` instead of `runner`? That seems to be the convention in Kotlin for passing around lambdas.